### PR TITLE
Fixed tests, tab spaces, and updated shared submodule ptr

### PR DESCRIPTION
**Pull request by eric-romero**

### CODE REVIEW

File: libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreExternalStorageTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		int dbBlobsDirSizeAfter = totalSizeAfter - dbFileSizeAfter;
+
+		assertTrue("Database file should be larger", dbFileSizeAfter > dBFileSizeBefore);
+		assertTrue("Soup blobs directory should be larger", dbBlobsDirSizeAfter > dbBlobsDirSizeBefore);
+		assertTrue("Total database size should be larger than just db file", totalSizeAfter > totalSizeBefore);
+	}
+
+	@Override
+	public void testAggregateQueryOnJSON1IndexedField() throws JSONException {
+		// json1 is not compatible with external storage.
```
Comment (wmathurin) [inline libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreExternalStorageTest.java]: Why have the test if it is not applicable?

Comment (wmathurin) [inline libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreExternalStorageTest.java]: Never mind, i just realized you are subclassing the other test class.

**Comment by wmathurin** [comment]
LGTM
I think we can merge it in - we have a post-merge story for refinements if needed
@bhariharan ?

**Comment by bhariharan** [comment]
Looks okay to me. One question for @eric-romero - are we removing the external stores per user on logout, like we remove the other ```SmartStore``` databases? I couldn't find the code for it in this PR.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java [hunk 2]
```diff
@@ -237,6 +258,11 @@ public static synchronized void deleteDatabase(Context ctx, String dbNamePrefix,
 				communityDBNamePrefix.append(accountSuffix);
 		    	deleteFiles(ctx, communityDBNamePrefix.toString());
 			}
+
+			// Delete external blobs directory
+			StringBuilder blobsDbPath = new StringBuilder(ctx.getApplicationInfo().dataDir);
+			blobsDbPath.append("/databases/").append(fullDBName).append(EXTERNAL_BLOBS_SUFFIX);
+			removeAllFiles(new File(blobsDbPath.toString()));
```
Comment (eric-romero) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: @bhariharan external storage is removed here in deleteDatabase along when the .db file itself is removed

Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: Cool, thanks.

**Comment by bhariharan** [comment]
👍  Merging this in, @wmathurin.

**Comment by wmathurin** [comment]
@eric-romero How do I change the storage type of a soup. In other words, where is the Android equivalent of https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/unstable/libs/SmartStore/SmartStore/Classes/SFSmartStore.m#L2107?

**Comment by eric-romero** [comment]
@wmathurin this is missing on android, i will try to put it together today

**Pull request merged**